tools/c7n_oci/c7n_oci/resources/compute.py

```diff
@@ -0,0 +1,296 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import oci.core
+
+from c7n.filters import Filter
+from c7n.utils import type_schema
+from c7n_oci.actions.base import OCIBaseAction, RemoveTagBaseAction
+from c7n_oci.provider import resources
+from c7n_oci.query import QueryResourceManager
+
+log = logging.getLogger('custodian.oci.resources.compute')
+
+@resources.register('instance')
+class Instance(QueryResourceManager):
+    """Oracle Cloud Infrastructure Instance Resource
+
+    :example:
+
+    Returns all Instance resources in the tenancy
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-all-instance-resources
+              resource: oci.instance
+
+    """
+
+    class resource_type:
+        doc_groups = ['Compute']
+        service = 'oci.core'
+        client = 'ComputeClient'
+        enum_spec = ('list_instances', 'items[]', None)
+        extra_params = {'compartment_id','instance_id'}
+        resource_type = 'OCI.Compute/Instance'
+        id = 'identifier'
+        name = "display_name"
+        search_resource_type = 'instance'
+
+
+    def _construct_query_params(self):
+        default_filters = super().get_default_filters()
+        ''' Only return the params that are eligible to be passed to the api, as defined in extra_params '''
+        default_filters = dict((k, v) for k, v in default_filters.items() if k in self.resource_type.extra_params)
+        return default_filters
+
+    def get_resources(self, params, enum_spec):
+        monitoring_filter = list(filter(lambda x: x.data.get('type') == 'monitoring', self.filters))
```

Comment:
this logic should move into the metrics filter instead of being in the resource manager. ideally we'd have a metrics filter base class in filters/ package as some of this will be common across resource types.